ui-performance.md

```diff
@@ -0,0 +1,470 @@
+Select **Flutter: Open Observatory Timeline** from the list that pops up.
+If this command isn’t available, make sure that the app is running.
+[PENDING: Provide a screenshot when the UI is stable.]
+</li>
+</ol>
+
+#### Using Observatory's timeline
+
+Observatory is a web application that is hosted by the VM. It provides a suite of
+program inspection and analysis tools, but we are primarily using the timeline to
+grab and view slices of the stack to determine where your Flutter app is spending
+its time.
+
+<ol markdown="1">
+
+<li markdown="1"> When bringing up Observatory from an IDE,
+    it takes you directly to the timeline view.
+    If you are looking at Observatory’s front page,
+    click the “view <ul>timeline</ul>” link.
+
+    Initially, timeline page contains no timeline data. In the **Recorded Streams Profile**,
+    select the **Flutter Developer profile** radio button. This simultaneously selects the **Dart**,
+    **Embedder**, and **GC** streams.
+</li>
+
+<li markdown="1"> Run the app and get it to the state that you want to examine.
+</li>
+
+<li markdown="1"> Click the **Refresh** button at the top right.
+    This grabs the stack at its current state
+    and displays a screen that looks something like the following:
+
+<center><img srw="images/timeline-initial.png" alt="Screenshot of the timeline with values from the running app"></center>
+
+<!--
+1. Each of the "hairs" on the timeline represent a time slice from the Dart profiler.
+You'll see slices for each phase: layout, compositing, animation, etc.
+-->
+
+    You can see data, but you can't read it. To make the data readable, hover the cursor over the
+    area in the view that you want to enlarge and use the keyboard keys (as many times as needed)
+    as follows:
+
+<ul markdown="1">
+<li markdown="1"> w - widen
+</li>
+<li markdown="1"> s - narrow
+</li>
+<li markdown="1"> a - shift right
+</li>
+<li markdown="1"> d - shift left
+</li>
+</ul>
+
+    Once enlarged, the bottom middle graph looks like the following:
+
+<center><img src="images/timeline-data-enlarged.png" "alt="Screenshot showing blocks of color with text labels"></center>
+
+    Click on one of the hairs and you'll see a stack trace with the corresponding
+    Dart code for that instant.
+    [PENDING: Get feedback from Ryan to make this more understandable.]
+</li>
+</ol>
+
+If the problem is in your Dart code, fix that performance issue.
+
+If the problem is buried deep in the framework's code, then the app is doing something
+to make the framework slow. You can learn more by adding custom slices to the timeline
+using the [Timeline](https://docs.flutter.io/flutter/dart-developer/Timeline-class.html)
+class in the
+[dart:developer](https://docs.flutter.io/flutter/dart-developer/dart-developer-library.html)
+package.
+
+[PENDING: would be nice to have example of code with time slices, but not required]
+
+Iterate until you can determine what in your code is causing the performance issue.
+
```

Comment:
@chinmaygarde can explain it better but my understanding is that there's a limit to how much timeline data we're storing in memory and when refreshing, only the last few seconds' data will be shown in Chrome tracing. We should describe it so people won't be wondering why their Timeline calls aren't appearing.